.github/workflows/autobuild.yaml

```diff
@@ -1,41 +1,125 @@
-# Workflow to build binaries for a release.
-# Triggered by release creation, which should include `llvm-project@<full-sha>`.
+# Workflow to build binaries and release them.
+# Triggered by the schedule or manual dispatch, which might include
+# `<owner/llvm-project>@<full-sha>`.
 #
 # Because the build takes more than an hour, our GITHUB_TOKEN credentials may
 # expire. A token `secrets.RELEASE_TOKEN` must exist with public_repo scope.
 name: Build release binaries
 on:
-  release:
-    types: created
+  # Run weekly on sunday at 21:37 UTC (arbitrary)
+  schedule:
+  - cron: '37 21 * * SUN'
+  workflow_dispatch:
+    inputs:
+      commit:
+        description: 'Commit to build from'
+        required: false
+      repo:
+        description: 'Repository to build from'
+        required: false
+        default: 'llvm/llvm-project'
+      tag:
+        description: 'Tag name'
+        required: false
+        default: ''
 jobs:
-  prepare:
+  # Choose the commit to build a release from.
+  #
+  # We want to avoid unbuildable revisions: choose the last green from CI.
+  # FIXME: The criteria should be some consistent set of buildbots passing.
+  pick:
+    name: Pick LLVM commit
     runs-on: ubuntu-latest
     steps:
-    - name: Find LLVM commit
+    - name: Install deps
+      run: |
+        sudo apt-get install jq
+    - name: Clone scripts
+      uses: actions/checkout@v2
+    - name: Pull repo from the inputs
+      if: ${{ github.event.inputs.repo == '' }}
       run: |
-        perl -ne 'print "LLVM_REPO=$1\nLLVM_COMMIT=$2\n" if /(\w+\/llvm-project)@([[:xdigit:]]{40,})/' > commit.env <<EOF
-        ${{ github.event.release.body }}
-        EOF
-    - name: Check LLVM commit was founud
-      run: grep LLVM_REPO commit.env
-    - name: Mark as draft
+        echo "LLVM_REPO=llvm/llvm-project" >> commit.env
+    - name: Use the default repo
+      if: ${{ github.event.inputs.repo != '' }}
+      run: |
+        echo "LLVM_REPO=${{ github.event.inputs.repo }}" >> commit.env
+    # Use clangd/actions/pick after
+    # https://github.com/ramasilveyra/last-successful-gh-commit/issues/2 has
+    # been addressed.
+    - name: Get commit hash for LLVM head
+      if: ${{ github.event.inputs.commit == '' }}
       run: >
-        curl --fail --show-error -XPATCH
-        "-HAuthorization: Bearer ${{ secrets.RELEASE_TOKEN }}"
-        "https://api.github.com/repos/${{ github.repository }}/releases/${{ github.event.release.id }}"
-        "-d" '{"draft": true}'
-    - name: Persist release info
-      uses: actions/upload-artifact@v1
+        COMMIT=$(curl --fail --show-error
+        "https://api.github.com/repos/llvm/llvm-project/commits/main" |
+        jq ".sha" -r)
+
+        echo "LLVM_COMMIT=$COMMIT" >> commit.env
+    - name: Push picked commit to the file
+      if: ${{ github.event.inputs.commit != '' }}
+      run: |
+        echo "LLVM_COMMIT=${{ github.event.inputs.commit }}" >> commit.env
+    - name: Upload result
```

Comment:
rather than uploading and downloading results as artifacts, i believe we can directly put them to `$GITHUB_ENV` since we have a single workflow now and they should be available to all subsequent actions. details https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable